Update all documentation for 2.7.0 (#7533)
keepassxreboot/keepassxc: docs/topics/DatabaseOperations.adoc

.Unlocked database
image::database_view.png[]

=== Quick Unlock

> On Windows and macOS, depending on hardware availability, KeePassXC will securely store your credentials to enable quickly unlocking your database after providing biometric authentication. This is enabled by default on Windows using _Windows Hello_ and on macOS using _Touch ID or Apple Watch_ services. You can disable this feature in the Application Settings under the Security section.

Reviewer: Should we say "KeePassXC will securely store your credentials" when in fact we are just asking the operating system to do so meaning the _security_ is provided by the OS and not us? Splitting hairs but noting.

Author: Maybe "Your credentials will be securely stored ..."

Reviewer 2: ```suggestion
If your device supports it, KeePassXC will securely store your credentials to enable quickly unlocking your database with biometric authentication via _Windows Hello_, _Apple Touch ID_, or your _Apple Watch_. This is feature is supported and enabled by default on Windows and macOS. You can disable it in the Application Settings under the Security section.
```

NOTE: On Windows you will be prompted to authenticate to Windows Hello on initial database unlock. This is required to access the secure certificate store providing the encryption key to store your credentials.

.Windows Hello example
image::quick_unlock_windows_hello.png[]

> When your database locks due to timeout, screen lid closure, or manually locking you will see the following unlock dialog. Simply press _Enter_ or click on _Unlock Database_ to initiate the biometric authentication to unlock your database. If you are using a Hardware Key, you must have the key inserted into your computer to complete the unlock.

Reviewer 2: ```suggestion
When your database is locked, you will see the following unlock dialog. Simply press _Enter_ or click on _Unlock Database_ to initiate the biometric authentication to unlock your database. If you are using a hardware key, you must have the key inserted into your computer to complete the unlock.
```
KISS

image::edit_entry.png[]

2. Enter a desired title for the entry, username, password, URL, and notes on this screen.

>    a. Your most frequently used usernames will automatically be available in the username drop down. They will also auto-complete for you when typing.

Reviewer 2: ```suggestion
   a. Your most frequently used usernames will automatically be available in the username drop-down menu. They will also auto-complete for you when typing.

```

   b. You can generate a secure random password by clicking the dice icon in the password field to launch the password generator. Reveal the password by clicking the eye icon.

>    c. After you add a URL to an entry you can press the download button to automatically download the website's favorite icon and set it on this entry.

Reviewer: "favorite icon" doesn't really mean much in 2022, maybe just call it icon?

> 3. _(Optional)_ Add tags to the entry so you can quickly search for it using the tags panel on the main database view. You can easily add new tags or select existing ones from the drop-down list.

Reviewer 2: ```suggestion
3. _(Optional)_ Add tags to the entry, so you can quickly search for it using the tags panel on the main database view. You can easily add new tags or select existing ones from the drop-down list.

```

> 4. _(Optional)_ Select the expires check-box to set the expiry date for the password. You can manually enter the date and time or click the Presets button to select an expiry date and time for your password.

Reviewer 2: ```suggestion
4. _(Optional)_ Select the _Expires_ check-box to set the expiry date for the password. You can manually enter the date and time or click the _Presets_ button to select an expiry date and time for your password.

```

4. Click *OK*.

=== Adding TOTP to an Entry

> Timed One-Time Passwords (TOTP) are a popular choice for two-factor authentication methods. These codes are typically six digits long and change every 30 seconds. They are derived from a shared secret value and the current time. Once set up, KeePassXC will calculate the current value of a TOTP code for use with copy/paste, browser extension, and Auto-Type.

Reviewer 2: ```suggestion
Timed One-Time Passwords (TOTP) are a popular choice for two-factor authentication. These codes are typically six digits long and change every 30 seconds. They are derived from a shared secret value and the current time. Once set up, KeePassXC can calculate TOTP codes like any TOTP authenticator app, such as Google Authenticator. The codes can be used by copy/paste, the browser extension, or Auto-Type.
```

WARNING: Your computer time must be synchronized with an internet time source to generate valid TOTP codes, https://www.nist.gov/pml/time-and-frequency-division/time-distribution/internet-time-service-its[read more here].

> 

Reviewer: Maybe we should add a note/warning that storing a TOTP secret with your password in the same database defeats the purpose of it, even when it's convenient?

Reviewer 3: I have had that thought too. Would it be possible / sensible to implement two secure stores inside a keepassxc file, ie. One for passwords as usual and a separate one which only contains the totp secrets, which must be unlocked by a separate unrelated password to be used ?

To add TOTP to a database entry, you must first retrieve the secret string from the website or application you are authenticating to. Often this secret is accompanied with a QR code and can be copy/pasted below. Example:

.Example TOTP Secret
image::totp_code_example.png[width=50%]

> Once obtained, right click the desired entry *(1)*, choose _TOTP_ -> _Set up TOTP..._ *(2)*, and the setup dialog will appear. In that dialog, paste the secret code from the website *(3)*, setup any custom settings (rare) *(4)*, then press OK to save the settings.

Reviewer 2: ```suggestion
Once obtained, right-click the desired entry *(1)*, choose _TOTP_ -> _Set up TOTP..._ *(2)*, and the setup dialog will appear. In that dialog, paste the secret code from the website *(3)*, setup any custom settings (rare) *(4)*, then press OK to save the settings.

```

.TOTP Setup Process
image::totp_setup.png[]

> After an entry is configured with TOTP you will see a clock icon in that entry's row and the ability to reveal the current code in the preview pane. Additionally, you can navigate to the entry's _TOTP_ menu and can show the code in a separate window or the QR code for exporting the configuration to a mobile device. This code can be used with the browser extension without any additional changes and with Auto-Type by using the `{TOTP}` placeholder or menu options in the Auto-Type selection dialog.

Reviewer 2: ```suggestion
After an entry is configured with TOTP, you will see a clock icon in that entry's row and have the ability to reveal the current code in the preview pane. Additionally, you can navigate to the entry's _TOTP_ menu to show the code in a separate window. You can also view the secret and configuration as a QR code for exporting it to a mobile device. TOTP codes can be entered into forms with the browser extension, with Auto-Type by using the `{TOTP}` placeholder, or via the menu options in the Auto-Type selection dialog.
```

>  * Argon2 (KDBX 4 - recommended): KDBX 4, the Argon2 key derivation function can be used for transforming the composite master key (as protection against dictionary attacks). The main advantage of Argon2 over AES-KDF is that it provides a better resistance against GPU/ASIC attacks (due to being a memory-hard function). The number of iterations scales linearly with the required time. By increasing the memory parameter, GPU/ASIC attacks become harder (and the required time increases). The parallelism parameter can be used to specify how many threads should be used.

Reviewer 4: Need to mention anything about Argon2d and Argon2id? We have those two options in Database Settings but no explanations of them.

Author: I thought about it, can easily add a sentence on the difference.

== Database Maintenance

> KeePassXC offers some maintenance features that can be applied to cleanup your database. 

Reviewer 2: ```suggestion
KeePassXC offers some maintenance features that can be applied to clean up your database. 
```

> 1.	Navigate to _Database_ -> _Database settings_. The following screen appears:

Author: Forgot to add the image and some text about purging unused icons.

```

You will be asked to enter the HMAC key you created earlier, copy/paste they key output in the first step. Repeat both steps for your second YubiKey. We recommend storing your HMAC key in a safe place (e.g., printed on paper) in case you need to recreate another key.

== Command Line Tool

> KeePassXC comes with a robust command line tool *keepassxc-cli* to access, view, and manipulate your database directly from any terminal window. The tool is separately documented through it's man page which can be shown using `man keepassxc-cli` or through the on-demand help using `keepassxc-cli [command] -h`. An online version of the man page is accessible https://github.com/keepassxreboot/keepassxc/blob/master/docs/man/keepassxc-cli.1.adoc[here].

Reviewer: "it's man page" -> "its man page".

Author: Possessive is correct here. Although I think we should say "the man page"

> KeePassXC comes with a robust command line tool *keepassxc-cli* to access, view, and manipulate your database directly from any terminal window. The tool is separately documented through it's man page which can be shown using `man keepassxc-cli` or through the on-demand help using `keepassxc-cli [command] -h`. An online version of the man page is accessible https://github.com/keepassxreboot/keepassxc/blob/master/docs/man/keepassxc-cli.1.adoc[here].

Reviewer 2: ```suggestion
KeePassXC comes with the command line tool *keepassxc-cli* to access, view, and manipulate your database directly from a terminal window. The tool is documented through a separate man page, which can be shown using `man keepassxc-cli`, or through the on-demand help using `keepassxc-cli [command] -h`. An online version of the man page is https://github.com/keepassxreboot/keepassxc/blob/master/docs/man/keepassxc-cli.1.adoc[available on GitHub].
```

Don't use _here_ links.

// end::advanced[]

== Storing a Database File
The database file that you create might contain highly sensitive data and must be stored in a very secure way. You must make sure that the database is always protected with a strong and long password. The database file that is protected with a strong and long password is secure and encrypted while stored on your computer or cloud storage service.

> Make sure that you or someone else does not accidentally delete the database file. Deletion of the database file will result in the total loss of your information and a lot of inconvenience to manually retrieve your logins for various web applications. We do not recommend sharing your database file with anyone unless you absolutely trust them (spouse, child, etc).

Reviewer: Sharing is not well defined here. I know you just modified the text a bit but I assume the idea here is sharing the same file between two or more people knowing the same shared secret to open it.

> Make sure that you or someone else does not accidentally delete the database file. Deletion of the database file will result in the total loss of your information and a lot of inconvenience to manually retrieve your logins for various web applications. We do not recommend sharing your database file with anyone unless you absolutely trust them (spouse, child, etc).

Reviewer 2: ```suggestion
Make sure that you or someone else does not accidentally delete the database file. Deletion of the database file will result in the total loss of all your information (including all your passwords!) and a lot of inconvenience to manually retrieve your logins for various web applications. Do not share the credentials to access your database file with anyone unless you absolutely trust them (spouse, child, etc.).
```

> TIP: It is safe to store your database file in cloud storage (e.g., OneDrive, Dropbox, Google Drive, etc). The file is always encrypted using your chosen encryption options. At no time is unencrypted data written to disk or accessible to your cloud storage provider. Using a cloud storage service enables automatic backup of your database file in the event of corruption or accidental deletion.

Reviewer: As sharing above was not well defined and it was not recommended but here we tell that it's safe to give the file to a third party. Splitting hairs again.

Author: Ehhh hairs have been split. Although I should say that "Most cloud storage providers provide automatic version history"

> TIP: It is safe to store your database file in cloud storage (e.g., OneDrive, Dropbox, Google Drive, etc). The file is always encrypted using your chosen encryption options. At no time is unencrypted data written to disk or accessible to your cloud storage provider. Using a cloud storage service enables automatic backup of your database file in the event of corruption or accidental deletion.

Reviewer 2: ```suggestion
TIP: It is safe to store your database file in the cloud (e.g., OneDrive, Dropbox, Google Drive, Nextcloud, Syncthing, etc), since the file is always fully encrypted. At no time is unencrypted data written to disk or becomes accessible to your cloud storage provider. Using a cloud storage service enables automatic backups of your database file in the event of corruption or accidental deletion.
```
